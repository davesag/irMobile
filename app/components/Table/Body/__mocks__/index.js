import React from 'react'
import { View, Text } from 'react-native'

export default jest.fn(({ data, makeKey }) => (
  <View>
    {data.map((d, i) => (
      <View key={d[0]}>
        {d.map((dd, j) => (
          <Text key={makeKey(d[0], j)}>{dd}</Text>
        ))}
      </View>
    ))}
  </View>
))
