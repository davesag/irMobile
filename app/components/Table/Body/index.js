import React from 'react'
import PropTypes from 'prop-types'
import { ScrollView } from 'react-native'

import Row from '../Row'

const Body = ({ data, makeKey }) => (
  <ScrollView>
    {data.map(row => (
      <Row
        key={`row-${row[0]}`}
        data={row}
        makeKey={(value, i) => makeKey(row[0], i)}
      />
    ))}
  </ScrollView>
)

Body.propTypes = {
  data: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
  makeKey: PropTypes.func.isRequired
}

Body.defaultProps = {
  data: null
}

export default Body
