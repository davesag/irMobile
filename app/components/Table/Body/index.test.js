import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Body from '.'

const body = [['A', 'B', 'C'], ['D', 'E', 'F']]
const makeKey = (value, i) => i

let tree

jest.mock('../Row')

describe('rendering', () => {
  beforeAll(() => {
    tree = renderer.create(<Body data={body} makeKey={makeKey} />)
  })

  it('rendered correctly ', () => {
    expect(tree).toMatchSnapshot()
  })
})
