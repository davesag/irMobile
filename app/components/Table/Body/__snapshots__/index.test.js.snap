// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering rendered correctly  1`] = `
<RCTScrollView>
  <View>
    <View>
      <Text>
        d
      </Text>
      <Text>
        d
      </Text>
      <Text>
        d
      </Text>
    </View>
    <View>
      <Text>
        d
      </Text>
      <Text>
        d
      </Text>
      <Text>
        d
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
